File: examples/typescript/end-users/getDelegationForAccount.ts
// Usage: pnpm tsx end-users/getDelegationForAccount.ts <USER_UUID> <ADDRESS>

import { CdpClient } from "@coinbase/cdp-sdk";
import "dotenv/config";

const userId = process.argv[2];
const address = process.argv[3];

if (!userId || !address) {
  console.error(
    "Usage: pnpm tsx end-users/getDelegationForAccount.ts <USER_UUID> <ADDRESS>",
  );
  process.exit(1);
}

const cdp = new CdpClient({
  basePath: process.env.CDP_BASE_PATH,
});

try {
  // Get the account-scoped delegation via the client method
  const delegation = await cdp.endUser.getDelegationForEndUserAccount({
    userId,
    address,
  });

  console.log("Account-scoped delegation via client method:", delegation);

  // Alternatively, use the EndUserAccount object shorthand
  const endUser = await cdp.endUser.getEndUser({ userId });

  const delegation2 = await endUser.getDelegationForAccount({ address });

  console.log("Account-scoped delegation via account method:", delegation2);
} catch (error) {
  console.error("Error: ", (error as { errorMessage: string }).errorMessage);
}

File: examples/python/end-users/get_delegation_for_account.py
"""Example: Get the active account-scoped delegation for an end user address.

Usage: python get_delegation_for_account.py <USER_UUID> <ADDRESS>
"""

import asyncio
import sys

from cdp import CdpClient


async def main(user_id: str, address: str) -> None:
    """Get an account-scoped delegation via client method and account shorthand."""
    async with CdpClient() as cdp:
        # Via the client method
        delegation = await cdp.end_user.get_delegation_for_end_user_account(
            user_id=user_id,
            address=address,
        )
        print(f"Delegation via client method: {delegation}")

        # Via the EndUserAccount shorthand
        end_user = await cdp.end_user.get_end_user(user_id=user_id)
        delegation2 = await end_user.get_delegation_for_account(address=address)
        print(f"Delegation via account method: {delegation2}")


if __name__ == "__main__":
    if len(sys.argv) != 3:
        print("Usage: python get_delegation_for_account.py <USER_UUID> <ADDRESS>")
        sys.exit(1)

    asyncio.run(main(sys.argv[1], sys.argv[2]))
